File: jar-common/src/main/java/com/beancounter/common/contracts/AssetRequest.java
package com.beancounter.common.contracts;

import com.beancounter.common.input.AssetInput;
import java.util.Map;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Singular;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class AssetRequest implements Payload<Map<String, AssetInput>> {
  @Singular("data")
  private Map<String, AssetInput> data;

}

File: jar-common/src/main/java/com/beancounter/common/contracts/AssetRequest.kt
package com.beancounter.common.contracts

import com.beancounter.common.input.AssetInput
import org.springframework.boot.context.properties.ConstructorBinding
import java.util.HashMap

/**
 * Contract to create a collection of assets.
 */
data class AssetRequest @ConstructorBinding
constructor(override var data: Map<String, AssetInput> = HashMap()) : Payload<Map<String, AssetInput>> {
    constructor(code: String, assetInput: AssetInput) : this(data = hashMapOf(code to assetInput))
}
